app/controllers/owners/unaccepted_data_requests_controller.rb

```diff
@@ -0,0 +1,23 @@
+module Owners
+  class UnacceptedDataRequestsController < Owners::ApplicationController
+
+    def index
+      render json: json_data
+    end
+
+    private
+
+    def json_data
+      current_owner.companies.map {|company|
+        {
+          id: company.id,
+          name: company.name,
+          unaccepted_data_requests: company.data_requests.unaccepted
+        }
+      }.filter {|item| 
+        item[:unaccepted_data_requests].length > 0
+      }
```

Comment:
```suggestion
      current_owner.companies
        .filter { |company| company.data_requests.unaccepted.size > 0 }
        .map do |company|
          {
            id: company.id,
            name: company.name,
            unaccepted_data_requests: company.data_requests.unaccepted
          }
      end
```
1. turn it around to be more efficient
2. Use `do ... end` syntax for multi line blocks.
I think one should also make sure to eager load the data requests using `.include(:data_requests)` but I am not 100% sure so I leave this up to you :grin: